syntax = "proto3";

package chimera.v1;

option go_package = "Chimera/server/api/runtime/v1;v1";

service RuntimeService {
  rpc RunAgent (RunAgentRequest) returns (stream RunAgentResponse) {}
  rpc SyncDataSource (SyncRequest) returns (SyncResponse) {}
}

// --- è¿è¡Œç›¸å…³ ---

message RunAgentRequest {
  string app_id = 1;
  string query = 2;
  string session_id = 3;
  string app_config_json = 4;
  repeated Message history = 5;
}

message Message {
  string role = 1;
  string content = 2;
}

message RunAgentResponse {
  // type: "thought" | "delta" | "reference" | "error" | "summary"
  string type = 1;
  string payload = 2;
  AgentMeta meta = 3;

  // ğŸ”¥ æ–°å¢ï¼šæ‰§è¡Œæ‘˜è¦ (ä»…åœ¨ type="summary" æ—¶å­˜åœ¨ï¼Œä¸”ä½œä¸ºæµçš„æœ€åä¸€å¸§)
  RunSummary summary = 4;
}

message AgentMeta {
  string node_name = 1;
  string trace_id = 2;
  int64 duration_ms = 3;
}

// ğŸ”¥ æ–°å¢ï¼šç»Ÿè®¡æ•°æ®ç»“æ„
message RunSummary {
  int32 total_tokens = 1;
  int32 prompt_tokens = 2;
  int32 completion_tokens = 3;
  int64 total_duration_ms = 4; // Python ä¾§è®¡ç®—çš„æ€»è€—æ—¶
  string final_status = 5;     // "success" | "failed"
}

// --- ETL ç›¸å…³ (ä¿æŒä¸å˜) ---
message SyncRequest {
  int64 kb_id = 1;
  int64 datasource_id = 2;
  string type = 3;
  string config_json = 4;
}

message SyncResponse {
  bool success = 1;
  string error_msg = 2;
  int32 chunks_count = 3;
  int32 page_count = 4;
}